packages/node/src/dev-server.ts

```diff
@@ -136,77 +147,146 @@ async function serializeRequest(message: IncomingMessage) {
+          // we can't easily show a meaningful stack trace
+          // so, stick to just the error message for now
+          event.respondWith(new Response(error.message, {
+            status: 500,
+            headers: {
+              'x-handled-from': 'edge-wrapper'
+            }
+          }));
+        }
+      })`;
+  } catch (error) {
+    // We can't easily show a meaningful stack trace from ncc -> edge-runtime.
+    // So, stick with just the message for now.
+    console.log(`Failed to instantiate edge runtime: ${error.message}`);
+    return undefined;
+  }
+}
 
-      let body;
+async function createEdgeRuntime(userCode: string | undefined) {
+  try {
+    if (!userCode) {
+      return undefined;
+    }
 
-      if (requestDetails.method !== 'GET' && requestDetails.method !== 'HEAD') {
-        body = Uint8Array.from(atob(requestDetails.body), c => c.charCodeAt(0));
-      }
+    const edgeRuntime = new EdgeRuntime({
+      initialCode: userCode,
+      extend: (context: Primitives) => {
+        Object.assign(context, {
+          __dirname: '',
+          module: {
+            exports: {},
+          },
+        });
+        return context;
+      },
+    });
 
-      let requestUrl = requestDetails.headers['x-forwarded-proto'] + '://' + requestDetails.headers['x-forwarded-host'] + requestDetails.url;
+    const server = await runServer({ runtime: edgeRuntime });
+    exitHook(server.close);
 
-      let request = new Request(requestUrl, {
-        headers: requestDetails.headers,
-        method: requestDetails.method,
-        body: body
-      });
+    return server;
+  } catch (error) {
+    // We can't easily show a meaningful stack trace from ncc -> edge-runtime.
+    // So, stick with just the message for now.
+    console.log(`Failed to instantiate edge runtime: ${error.message}`);
+    return undefined;
+  }
+}
 
-      event.request = request;
-
-      let edgeHandler = module.exports.default;
-      let response = edgeHandler(event.request, event);
-      return event.respondWith(response);
-    })`;
-
-  const edgeRuntime = new EdgeRuntime({
-    initialCode,
-    extend: (context: Primitives) => {
-      Object.assign(context, {
-        __dirname: '',
-        module: {
-          exports: {},
-        },
-      });
-      return context;
-    },
-  });
-  const server = await runServer({ runtime: edgeRuntime });
-  exitHook(server.close);
+async function createEdgeEventHandler(
+  entrypoint: string
+): Promise<(request: IncomingMessage) => Promise<VercelProxyResponse>> {
+  const userCode = await compileUserCode(entrypoint);
+  const server = await createEdgeRuntime(userCode);
 
   return async function (request: IncomingMessage) {
+    if (!server) {
+      // this error state is already logged, but we have to wait until here to exit the process
+      // this matches the serverless function bridge launcher's behavior when
+      // an error is thrown in the function
+      process.exit(1);
```

Comment:
There's some dancing around being done at a few points in this file to make sure that errors are not thrown during setup and `process.exit` is called during `onDevRequest` instead. This maintains the interface of how the spawned dev server interacts with `startDevServer`.

We can probably refactor this to make it more obvious how these processes interact, but I wanted to see if I could get this to work with the same interface first. The good news is that this works as expected.